tests/remotes/gdrive.py

```diff
@@ -1,41 +1,85 @@
 # pylint:disable=abstract-method
+import locale
 import os
+import pathlib
+import tempfile
 import uuid
 
 import pytest
 from funcy import cached_property
 
-from dvc.path_info import CloudURLInfo
-from dvc.tree.gdrive import GDriveTree
+from dvc.path_info import CloudURLInfo, PathInfo
+from dvc.tree.gdrive import GDriveTree, GDriveURLInfo
 
 from .base import Base
 
 TEST_GDRIVE_REPO_BUCKET = "root"
 
 
 class GDrive(Base, CloudURLInfo):
+    def __init__(self, url, dvc):
+        CloudURLInfo.__init__(self, url)
+        Base.__init__(self, url)
+        self.tree = GDriveTree(dvc, self.config)
+
```

Comment:
Maybe just
```suggestion

    @cached_property
    def tree(self):
        return GDriveTree(dvc, self.config)

```
? Same as in other fixtures.

Reply:
I need to pass `dvc` to the class, as it comes from `tmp_dir.dvc`. And `tree` is public.

https://github.com/iterative/dvc/blob/689a569246634ed1f8367b7a0359e334964f1252/tests/remotes/gdrive.py#L90-L93

Reply:
@julsal Would it make sense to turn it around? The mkdir below is a hack, we didn't need tree before anywhere so we've initialized it there. Now we need the tree here, first and foremost, so it makes sense utilize the tree in the hack and not vice versa.